protostar/commands/test/cheatcodes/cheatcode.py

```diff
@@ -11,30 +11,38 @@
 from starkware.starknet.core.os.syscall_utils import BusinessLogicSysCallHandler
 from starkware.starknet.storage.starknet_storage import BusinessLogicStarknetStorage
 
-from protostar.commands.test.starkware.cheatable_carried_state import (
-    CheatableCarriedState,
-)
 from protostar.commands.test.starkware.cheatable_starknet_general_config import (
     CheatableStarknetGeneralConfig,
 )
+from protostar.utils.data_transformer_facade import DataTransformerFacade
 
 if TYPE_CHECKING:
     from protostar.commands.test.starkware.cheatable_execute_entry_point import (
         CheatableExecuteEntryPoint,
     )
+    from protostar.commands.test.starkware.cheatable_state import CheatableCarriedState
 
 
 class Cheatcode(BusinessLogicSysCallHandler):
+    """
+    #### ACHTUNG:
```

Comment:
Yea, we shouldn't use `__subclassess__`, problem is it has to be loaded before so python knows it exists. Can we just list Cheatcodes explicilty in `execute_entry_point`?

> The connection between `CheatcodeFactory::build` and `ConcreteCheatcode::constructor` is not type-safe.

I wouldn't write such stuff in docstring. Python generally is not typesafe and nobody should rely on any type guarantees. It's unnecessary information which may be confusing

Reply:
| nobody should rely on any type guarantees

Unless the project uses types quite heavily. If you can't trust types, then the function shouldn't use it or use `Any` type.

Anyway, this will be removed.

Reply:
Unless the project uses types quite heavily. If you can't trust types, then the function shouldn't use it or use Any type.

Yep exactly, you should store such information using type hints, not docstrings